src/mlpack/methods/ann/layer/elu.hpp

```diff
@@ -210,7 +210,7 @@ class ELU
    */
   double Deriv(const double y)
   {
-    return (y > 0) ? lambda : lambda * (y + alpha);
+    return (y > 0) ? lambda : lambda * (Fn(y) + alpha);
```

Comment:
My derivation suggests that `lambda * (Fn(y) + alpha)` is incorrect.  Instead, for x < 0,

```
f(x) = lambda * alpha * (e^x - 1);

f'(x) = lambda * alpha * e^x;
       = lambda * alpha * e^x - lambda * alpha + lambda * alpha;
       = f(x) + lambda * alpha;
```

which is a bit different than what you've got here.  But I do agree that the original implementation was not correct (and actually I think the LaTeX at the top of the class is also not correct).

Reply:
*Duh* I should have derived it instead of going with the comments. But yes, as you said the earlier implementation was incorrect.

Reply:
For the ELU function ``lambda`` is set to 1: 

https://github.com/mlpack/mlpack/blob/713b7c678fce8dafab4a01905a8e4b4a56ca1bac/src/mlpack/methods/ann/layer/elu.hpp#L115-L123

see the note "Use this constructor for ELU activation function.".

so the ``f(x)' = f(x) + alpha;``, also note y is already the value of ``f(x)``.

Reply:
@zoq then the condition y > 0 in the derivative function makes no sense.

Reply:
Oh, I see, if `y == f(x)` then the original implementation seems to be correct.  But I don't see what will call `Deriv()` with `f(x)` instead of `x`... it doesn't look like `Deriv(const InputType& x, OutputType& y)` does this.  Maybe I overlooked something?

Reply:
The ELU ``Backward`` method will call the function with: ``Deriv(f(x), OutputType& y)`` to avoid an unnecessary recalculation.

> @zoq  then the condition y > 0 in the derivative function makes no sense.
True, we should definitely fix that.

Reply:
Ah, got it.  My opinion is that we should add some documentation to the `Deriv()` function then to point that out. :)